score-lang-runtime/src/main/java/org/openscore/lang/runtime/bindings/LoopsBinding.java

```diff
@@ -86,10 +88,17 @@ private LoopCondition createForLoopCondition(ForLoopStatement forLoopStatement,
             throw new RuntimeException("Error evaluating for loop expression in task '" + nodeName + "', error is: \n" + t.getMessage(), t);
         }
 
-        if ((forLoopStatement instanceof MapForLoopStatement) && !(evalResult instanceof PyList)) {
-            throw new RuntimeException(
-                    "Invalid expression for iterating maps: " + collectionExpression
-                    + "\nValid format is: map.items()");
+        if (forLoopStatement instanceof MapForLoopStatement) {
+            if (evalResult instanceof Map) {
+                List<Map.Entry<Serializable, Serializable>> entriesAsSerializable = Lists.newArrayList();
+                @SuppressWarnings("unchecked") Set<Map.Entry<Serializable, Serializable>> entrySet = ((Map) evalResult).entrySet();
+                for (Map.Entry<Serializable, Serializable> entry : entrySet) {
```

Comment:
We have to stay with Pair elements, because `new ArrayList(entrySet)` returns an object that is not Serializable